compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2449,6 +2453,34 @@ enum IsPattern {
     No,
 }
 
+/// Whether a binding is part of a use statement. Used for diagnostics.
+enum IsImport {
+    Yes,
+    No,
+}
+
+pub(crate) fn import_candidates(
+    session: &Session,
+    source_span: &IndexVec<LocalDefId, Span>,
+    err: &mut Diagnostic,
+    // This is `None` if all placement locations are inside expansions
+    use_placement_span: Option<Span>,
+    candidates: &[ImportSuggestion],
+) {
+    show_candidates(
+        session,
+        source_span,
+        err,
+        use_placement_span,
+        candidates,
+        Instead::Yes,
+        FoundUse::Yes,
+        IsPattern::No,
+        IsImport::Yes,
```

Comment:
Are these mutually exclusive? If so, they probably should be unified into one enum

Reply:
These two parameters cannot be `Yes` at the same time, but can be `No` at the same time, or one of them is `Yes`, so if they are unified into one enumeration, at least three variants are required.

Reply:
Yeah, that's what I meant -- the Yes+Yes state being invalid would be better represented as an enum.

Reply:
That makes sense. I think we can rename `IsPattern` enum to `Mode` and add a new variant in it. Since this PR is already in a rollup, should I just modify this PR or submit a new one?

Reply:
A follow-up is fine.